### ISSUE TITLE
MapGetWithNotNullAssertionOperator does not report violations on `MutableMap`s

### ISSUE THREAD

**Issue by travisMiehm** [detekt/detekt#9064]
## Expected Behavior
When `MapGetWithNotNullAssertionOperator` is enabled, I expect `mutableMapOf("one" to 1)["one"]!! to cause a violation to be reported.

I also expect this test to pass
```
    @Test
    fun `reports map_get() given a MutableMap`() {
        val code = """
            fun f() {
                val map = mutableMapOf<String, String>()
                map.get("key")!!
            }
        """.trimIndent()
        assertThat(subject.lintWithContext(env, code)).hasSize(1)
    }
```

## Observed Behavior
A violation is not reported, the test above fails with `Expected size: 1 but was: 0`

## Steps to Reproduce
See the example test above

## Context
We had an NPE in a production environment caused by somebody using the non-null get syntax. We relied on the rule to make sure people used `getValue()` which throws an exception that's better for troubleshooting.

## Your Environment
The example test fails on my fresh checkout of the main branch from earlier today.

**Comment by BraisGabin** [comment]
I guess, for your wording, that with `mapOf` it works. Only fails with the mutable one. Am I right? 

And, do you want to give it a try to this issue and send a pr?

**Comment by travisMiehm** [comment]
Correct, the issue is only with mutable maps.

I will send a PR, I think it's a relatively straightforward fix
